### PR TITLE
bpf: nodeport: consolidate packet rewrite in RevDNAT path

### PR DESCRIPTION
**Pull request by julianwiedmann**
This adds some smarts to the nodeport helpers, so that it's possible to split up the RevDNAT lookup (what IP/port to use for the RevDNAT) and the actual packet rewrite. This allows us to use the same helper code for the rewrite, regardless whether the reply comes from a local or a DSR backend.

It's a nice simplification on its own, but will also enable a follow-on fix in the RevDNAT path (https://github.com/cilium/cilium/pull/26638) that would otherwise be rejected by the verifier. So we'll backport this PR accordingly, once that fix is ready to merged.

### CODE REVIEW

**Comment by julianwiedmann** [comment]
https://github.com/cilium/cilium/pull/26638 needs a few more cycles until it's ready for primetime. But this part is good on its own already, so let's get started on review.

File: bpf/lib/conntrack.h [hunk 1]
```diff
@@ -1042,10 +1042,19 @@ ct_has_nodeport_egress_entry4(const void *map,
...
+		return false;
 
-	return 0;
+	if (entry->node_port) {
+		if (rev_nat_index)
+			*rev_nat_index = entry->rev_nat_index;
+		return true;
+	}
+
+	if (check_dsr && entry->dsr)
```
Comment (aspsk) [inline bpf/lib/conntrack.h]: Maybe just `return check_dsr && entry->dsr;` ?

File: bpf/lib/conntrack.h [hunk 2]
```diff
@@ -1076,10 +1085,19 @@ ct_has_nodeport_egress_entry6(const void *map,
...
+		return true;
+
+	return false;
```
Comment (aspsk) [inline bpf/lib/conntrack.h]: The code is identical with the v4 case. Maybe you can wrap the error checking and saving the `rev_nat_index` in an inline function?

**Review by aspsk — APPROVED** [review]

File: bpf/lib/conntrack.h (hunk 2 above)
Comment (julianwiedmann) [reply bpf/lib/conntrack.h]: I like it :+1: See the pushed diff - two birds, one stone.

**Pull request merged**